### DISCUSSION TITLE
IR-SOC Curve (parameters: LGM50_Chen2020)

### DISCUSSION THREAD

**Discussion by Mrzhang-hub** [pybamm-team/PyBaMM#1984] [Q&A] [answered]
As we all know, the internal resistance 'Local ECM resistance [Ohm]' will change while charging or discharging. I wanna ask have you conducted the experiment to find out the relationship between internal resistance and SOC. I have been writing an essay which use the parameters of LGM50_Chen2020 but I need the Internal Resistance-SOC curve which I could not find in that parameter files. Look forward to your reply!

**Comment by brosaplanella** [comment, marked as answer]
Hi! The internal resistance does not directly appear in the physics-based models, but instead, it is a concept defined to include all of the effects that cause a voltage loss in the battery. You can calculate the internal resistance by calculating the `Terminal voltage [V]` - `Measured open circuit voltage [V]` and dividing it by the applied current. **EDIT:** this is already calculated by the variable `"Local ECM resistance [Ohm]"` (see @tinosulzer's reply below).

**Reply by valentinsulzer** [reply]
> You can calculate the internal resistance by calculating the Terminal voltage [V] - Measured open circuit voltage [V] and dividing it by the applied current.

This is what "Local ECM resistance [Ohm]" is

**Reply by Mrzhang-hub** [reply]
Thank you for your clear explanation.

**Reply by Elmiar0642** [reply]
how to set/print the "Local ECM resistance [Ohm]" variable from `DFN()` model?

**Reply by valentinsulzer** [reply]
Read our examples to see how to get outputs from the model: https://docs.pybamm.org/en/stable/source/examples/notebooks/getting_started/tutorial-6-managing-simulation-outputs.html

Simple example:
```
import pybamm

model = pybamm.lithium_ion.DFN()  # Doyle-Fuller-Newman model
sim = pybamm.Simulation(model)
sol = sim.solve([0, 3600])
print(sol["Local ECM resistance [Ohm]"].data)
```

**Reply by Elmiar0642** [reply]
I found it myself after replying here by analysing the scripts. Now, could you please help me to set my own `Local ECM resistance [Ohm]` Values before starting a simulation?

**Reply by valentinsulzer** [reply]
You can't set that directly, it's an output of the model from a combination of various internal resistances (diffusion, kinetics, etc). If this is the kind of thing you want to do, you may want to use one of our (or your own) equivalent circuit models instead

**Reply by Elmiar0642** [reply]
I get it. I actually want to know where exactly they set the "Local ECM resistance [Ohm]" value (18th index in list passed) to the dictionary and with which particular value or culmination of values, to it? I tried searching throughout simulation.py and solution.py, I couldn't find it. Any hint should be a huge help!

**Reply by valentinsulzer** [reply]
Here: https://github.com/pybamm-team/PyBaMM/blob/develop/src/pybamm/models/full_battery_models/base_battery_model.py#L1443
